### ISSUE TITLE
Example C program using Babeltrace CTF writer library?

### ISSUE THREAD

**Issue by jstraus59** [efficios/babeltrace#12]
I am looking into generating CTF files directly from a Virtual Platform simulator that runs a wide variety of baremetal apps and OSes on a wide variety of processors. I want to use the Babeltrace CTF writer library in the simulator (which is written in C) and am trying to find an example of a C program that is using the CTF writer library. I see examples using the Python bindings but nothing  from C.

I have looked at [barectf](https://github.com/efficios/barectf) but it does not use the babeltrace library and also requires that the manifest is defined before compiling, while my manifest needs to be generated dynamically, by extracting trace event definitions from the virtual platform models.

I also note that if I try to use `babeltrace -o ctf` I get the message: `[error] Opening CTF traces for output is not supported yet.`

So I have 2 questions:

1) What is the status of the Babeltrace writer library?
2) Are there any examples of the library being used in a standalone C program that writes CTF files?

Thanks

Jim Straus

**Comment by jgalar** [comment]
Hi Jim,

The library has been available since Babeltrace v1.2. Babeltrace's next version will introduce a couple of additions to this library, but no breaking changes.

We're currently working on improving Babeltrace's documentation. In the meantime, the best example of using the CTF Writer library is probably the perf-to-ctf converter which is set to land in Linux 4.1.

https://github.com/torvalds/linux/blob/6c8a53c9e6a151fffb07f8b4c34bd1e33dddd467/tools/perf/util/data-convert-bt.c

The CTF Writer test suite might also help although the checks make the code hard to follow.
https://github.com/efficios/babeltrace/blob/master/tests/lib/test_ctf_writer.c

Don't hesitate to post any questions to the lttng mailing list: lttng-dev@lists.lttng.org

Regards,
Jérémie

**Comment by jstraus59** [comment]
Hi Jérémie,

That is exactly what I was looking for. Thanks!

Jim
